src/ui/server/helpers/get-from-session-or-application/index.ts

```diff
@@ -0,0 +1,28 @@
+import { RequestSessionUser, Application } from '../../../types';
+import { objectHasKeysAndValues } from '../object';
+
+/**
+ * getFromSessionOrApplication
+ * checks if application contains data for certain section and if it does, then returns that section and field
+ * if application does not contain relevant data, returns session
+ * @param {Application} application
+ * @param {String} section in application - eg. business
+ * @param {String} field eg. businessContactDetails
+ * @param {RequestSessionUser} session
+ * @returns {Object} session or application section
+ */
+const getFromSessionOrApplication = (application: Application, section: string, field: string, session?: RequestSessionUser) => {
+  if (!application[section] || !application[section][field]) {
+    return session;
+  }
+
+  const { id, __typename, ...sectionObject } = application[section][field];
```

Comment:
Can we add a condition and test to make sure `application[section]` exists? Currently if it doesn't this will error

Reply:
line 15.  It should return the session if application[section] does not exist.

And ts complains about the `submittedData `in `RequestSession` which is missing when I pass it req.session.user

Reply:
OK makes sense, sorry didn't see that.

RE the session in this case can we just rename `session` to `userSession`? and the file name as `getFromUserSessionOrApplication` just so it's clearer